controllers/cache/readiness.go

```diff
@@ -0,0 +1,257 @@
+package cache
+
+import (
+	"errors"
+	"fmt"
+	rediscli "github.com/go-redis/redis"
+	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/controllers/k8s"
+	"github.com/goharbor/harbor-cluster-operator/lcm"
+	corev1 "k8s.io/api/core/v1"
+	kerr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+const (
+	HarborChartMuseum = "chartmuseum"
+	HarborClair       = "clair"
+	HarborJobService  = "jobservice"
+	HarborRegistry    = "registry"
+)
+
+var (
+	components = []string{
+		HarborChartMuseum,
+		HarborClair,
+		HarborJobService,
+		HarborRegistry,
+	}
+)
+
+// Readiness reconcile will check Redis sentinel cluster if that has available.
+// It does:
+// - create redis connection pool
+// - ping redis server
+// - return redis properties if redis has available
+func (redis *RedisReconciler) Readiness() error {
+	var (
+		client *rediscli.Client
+		err    error
+	)
+
+	switch redis.HarborCluster.Spec.Redis.Kind {
+	case "external":
+		client, err = redis.GetExternalRedisInfo()
+	case "inCluster":
+		client, err = redis.GetInClusterRedisInfo()
+	}
```

Comment:
Do we add validation to the kind value? If there are some misspelling, is it possible to avoid such cases?

If not, maybe we need to add default here to return an error.

Reply:
we are validation value in the `harborcluster_types.go` 
```golang
// +kubebuilder:validation:Enum=inCluster;external
```